Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -138,28 +309,37 @@ void DataReaderTask::run(ProcessingContext& pc)
         } else { // we have compressed data coming in from flp.
           mCompressedReader.setDataBuffer(payloadIn);
           mCompressedReader.setDataBufferSize(payloadInSize);
-          mCompressedReader.configure(mByteSwap, mVerbose, mHeaderVerbose, mDataVerbose);
+          mCompressedReader.configure(mOptions);
           mCompressedReader.run();
         }
       } // ignore the input of DISTSUBTIMEFRAMEFLP
-        //      auto inputprocessingtime = std::chrono::high_resolution_clock::now() - inputprocessingstart;
-        //     LOGP(info, "Input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : processed in {} us",
-        //           dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize,std::chrono::duration_cast<std::chrono::microseconds>(inputprocessingtime).count());
+      //      auto inputprocessingtime = std::chrono::high_resolution_clock::now() - inputprocessingstart;
+      //     LOGP(info, "Input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : processed in {} us",
+      //           dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize,std::chrono::duration_cast<std::chrono::microseconds>(inputprocessingtime).count());
+    }
+
+    std::chrono::duration<double, std::milli> dataReadTime = std::chrono::high_resolution_clock::now() - dataReadStart;
+    LOG(info) << "Processing time for Data reading  " << std::chrono::duration_cast<std::chrono::milliseconds>(dataReadTime).count() << "ms";
+    if (mRootOutput) {
+      mTimeFrameTime->Fill((int)std::chrono::duration_cast<std::chrono::milliseconds>(dataReadTime).count());
+    }
+    //auto timeframe=mReader.getEventStorage();
+    // (*timeframe)->mTFStats.mTimeTaken= std::chrono::duration_cast<std::chrono::milliseconds>(dataReadTime).count();
+    // (*timeframe)->mTFStats.mDigitsFound = mReader.getDigitsFound();
+    // (*timeframe)->mTFStats.mTrackletsFound = mReader.getTrackletsFound();
+    //  (*timeframe)->mTFStats.mDataWordsRead = mWordsRead * 4;
+    // (*timeframe)->mTFStats.mDataWordsRejected = mWordsRejected * 4;
+    if (!mCompressedData) {
+      LOG(info) << "Digits found : " << mReader.getDigitsFound();
+      LOG(info) << "Tracklets found : " << mReader.getTrackletsFound();
+      LOG(info) << "DataRead in :" << mWordsRead * 4 << " bytes";
+      LOG(info) << "DataRejected in :" << mWordsRejected * 4 << " bytes";
+      LOG(info) << "DataRetention :bad/good" << (double)mWordsRejected / (double)mWordsRead << "";
+      LOG(info) << "Total % good data bad/(good+bad)" << (double)mWordsRejected / ((double)mWordsRead + (double)mWordsRejected) * 100.0 << " %";
```

Comment:
Did you look at this info? It looks like this should also show the inconsistencies between number of digits and tracklets at the link (?) level.

Reply:
Originally that was there, tracklets < digits/3 but then with no digits its flagged as error so removed it.